lib/rabbit_feed/connection.rb

```diff
@@ -0,0 +1,22 @@
+module RabbitFeed
+  class Connection
+    include Singleton
+
+    def initialize
+      @connection = Bunny.new RabbitFeed.configuration.connection_options
+      @connection.start
+      @channel = @connection.create_channel
+      @mutex = Mutex.new
+    end
+
+    protected
+
+    attr_reader :channel
+
+    def thread_safe &block
```

Comment:
Looking further down (to the indefinite `sleep`) it looks like no other Thread will ever get access to the object in that method (produce or consume), meaning that we could have some blocked threads here waiting for a lock they will never get. Is this a concern?. and if it is...
Shouldn't this be handled with something like (but better as this one still has a tiny window where it can return false for 2 threads)

```
if @mutex.locked?
  return
end

```